### ISSUE TITLE
Feature request: punch n' roll 

### ISSUE THREAD

**Issue by nesrad** [MaartenBaert/ssr#621]
Punch n' roll is used when you make a mistake and would like to quickly back up the recording, play back a few seconds, and start the recording again just before the mistake. This is particularly useful when recording presentations. I don't know how difficult this would be to implement with SSR, or if it's even possible, considering the recordings are encoded on-the-fly.

**Comment by MaartenBaert** [comment]
This is not possible without buffering significant amounts of uncompressed video, which would consume far too much RAM. Alternatively it would require rolling back the state of the encoder, or somehow resetting the state of the encoder by introducing a keyframe, but this is not something which the FFmpeg API supports. It's also highly format-dependent. I don't think this is practical.

**Comment by kn00tcn** [comment]
recording would have to be in chunks and/or the rollback can only align to keyframes

the user worflow described sounds almost the same as stopping the recording, starting a new one, then merging the files in an editor after cutting off the mistake at the end of the first file, so why bother implementing it in a 'simple' screen recorder

**Comment by MaartenBaert** [comment]
True, it probably makes more sense to just mark the timestamps where you need to do editing later, and then cut those out in a post-processing step. But that's still out of scope for SSR, I'm not going to make a video editor.

**Comment by kn00tcn** [comment]
'editor' is a strong word, simple cutting maybe joining is all one would want, but that's already here https://github.com/ozmartian/vidcutter

**Comment by hamoid** [comment]
I found this issue when looking for ways to avoid doing any editing of my video tutorials. I would often record 20 minutes then spend at least 1 hour cutting away the bad parts to leave 10 minutes.

I think two features that could be added to SSR that would make this easy, but I could also add such features via command line scripts. The features would be: to play the last recorded segment, and to delete said segment.

I think that would remove the need to edit the tutorials, as I can keep trying to shoot segments until I'm happy with them, otherwise discard. At the end I only need to concatenate all the segments that survived into one long video. What do you think of this approach? I need to try it...

**Comment by kn00tcn** [comment]
hmm, delete last segment sounds like an interesting hotkey idea

personally i would just open the file manager & delete, either right after the bad take or in the end, maybe create a small segment so that i'd know to delete the previous file afterward
